controllers/idler/idler_controller.go

```diff
@@ -198,6 +181,58 @@ func (r *Reconciler) ensureIdling(ctx context.Context, idler *toolchainv1alpha1.
 	return r.updateStatusPods(ctx, idler, newStatusPods)
 }
 
+// Check if the pod belongs to a controller (Deployment, DeploymentConfig, etc) and scale it down to zero.
+// if it is a standalone pod, delete it.
+// Send notification if the deleted pod was managed by a controller, was a standalone pod that was not completed or was crashlooping
+func deletePodsAndCreateNotification(ctx context.Context, podLogger logr.Logger, pod corev1.Pod, r *Reconciler, idler *toolchainv1alpha1.Idler) error {
+	lctx := log.IntoContext(ctx, podLogger)
```

Comment:
This is needed only when you need to propagate some values from the logger to the context, but I don't see anything like that here.
If you want to propagate the values set in the logger before, then it should be done outside of this function

Reply:
lctx is used in the scaleControllerToero function. It was called outside this function before. I also see that this change was purposely made in https://github.com/codeready-toolchain/member-operator/pull/495.
I've now passed two contexts to the function deletePodsAndCreateNotification in sync with the changes of the mentioned PR. Let me know what you think

Reply:
two contexts passed in - [9164efe](https://github.com/codeready-toolchain/member-operator/pull/630/commits/9164efed1c4569ec9780def2f3ef716fc2da1abe)